diff --git a/build/esm/platform/node/http-exporter-transport.js b/build/esm/platform/node/http-exporter-transport.js
index ef26668..36fc4bf 100644
--- a/build/esm/platform/node/http-exporter-transport.js
+++ b/build/esm/platform/node/http-exporter-transport.js
@@ -55,25 +55,22 @@ var HttpExporterTransport = /** @class */ (function () {
         this._send = null;
         this._agent = null;
     }
-    HttpExporterTransport.prototype.send = function (data, timeoutMillis) {
-        return __awaiter(this, void 0, void 0, function () {
-            var _a, sendWithHttp, createHttpAgent;
-            var _this = this;
-            return __generator(this, function (_b) {
-                if (this._send == null) {
-                    _a = require('./http-transport-utils'), sendWithHttp = _a.sendWithHttp, createHttpAgent = _a.createHttpAgent;
-                    this._agent = createHttpAgent(this._parameters.url, this._parameters.agentOptions);
-                    this._send = sendWithHttp;
-                }
-                return [2 /*return*/, new Promise(function (resolve) {
-                        var _a;
-                        // this will always be defined
-                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-                        (_a = _this._send) === null || _a === void 0 ? void 0 : _a.call(_this, _this._parameters, _this._agent, data, function (result) {
-                            resolve(result);
-                        }, timeoutMillis);
-                    })];
-            });
+    HttpExporterTransport.prototype.send = async function (data, timeoutMillis) {
+        if (this._send == null) {
+            // Lazy require to ensure that http/https is not required before instrumentations can wrap it.
+            const { sendWithHttp, createHttpAgent,
+            // eslint-disable-next-line @typescript-eslint/no-var-requires
+             } = await import('./http-transport-utils');
+            this._agent = createHttpAgent(this._parameters.url, this._parameters.agentOptions);
+            this._send = sendWithHttp;
+        }
+        return new Promise(resolve => {
+            var _a;
+            // this will always be defined
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            (_a = this._send) === null || _a === void 0 ? void 0 : _a.call(this, this._parameters, this._agent, data, result => {
+                resolve(result);
+            }, timeoutMillis);
         });
     };
     HttpExporterTransport.prototype.shutdown = function () {
diff --git a/build/esnext/platform/node/http-exporter-transport.js b/build/esnext/platform/node/http-exporter-transport.js
index e6b76f3..32a6244 100644
--- a/build/esnext/platform/node/http-exporter-transport.js
+++ b/build/esnext/platform/node/http-exporter-transport.js
@@ -24,7 +24,7 @@ class HttpExporterTransport {
             // Lazy require to ensure that http/https is not required before instrumentations can wrap it.
             const { sendWithHttp, createHttpAgent,
             // eslint-disable-next-line @typescript-eslint/no-var-requires
-             } = require('./http-transport-utils');
+             } = await import('./http-transport-utils');
             this._agent = createHttpAgent(this._parameters.url, this._parameters.agentOptions);
             this._send = sendWithHttp;
         }
