diff --git a/dist/index.js b/dist/index.js
index e798670e0bcee619c181680665eb4d8755f49b07..09b8df1bbba3ffdb0c464ff62b9c0b3926e85cb3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1717,7 +1717,7 @@ async function saveSnapshotFile(environment, snapshotData, snapshotPath) {
       normalizeNewlines(snapshotData[key])
     )};`
   );
-  const content = `${environment.getHeader()}
+  const content = `// Jest Snapshot v1, https://goo.gl/fbAQLP
 
 ${snapshots.join("\n\n")}
 `;
@@ -2201,10 +2201,10 @@ class SnapshotClient {
         throw err;
       }
     }
-    const testName = [name, ...message ? [message] : []].join(" > ");
+    const testName = [name, ...message ? [message] : []].join(": ");
     const { actual, expected, key, pass } = snapshotState.match({
       testId,
-      testName,
+      testName: testName.replaceAll(' >', ''),
       received,
       isInline,
       error,
