diff --git a/index.js b/index.js
index 2e8e916e79d66cd401fea3ca3653fc96a1703b83..e7dc1ce9880041b2e58ca57bc7ba8bc86c27d9ca 100644
--- a/index.js
+++ b/index.js
@@ -80,7 +80,12 @@ function Converter (sm, opts) {
   }
 
   if (opts.isJSON || opts.encoding) {
-    sm = JSON.parse(sm);
+    try {
+      sm = JSON.parse(sm);
+    } catch(e) {
+      console.error('Could not parse source map', sm, e);
+      return;
+    }
   }
 
   this.sourcemap = sm;
