diff --git a/jsutils/instanceOf.js b/jsutils/instanceOf.js
index 18403cf..61ce6be 100644
--- a/jsutils/instanceOf.js
+++ b/jsutils/instanceOf.js
@@ -44,6 +44,7 @@ const instanceOf =
               : _value$constructor.name;
 
           if (className === valueClassName) {
+            return true;
             const stringifiedValue = (0, _inspect.inspect)(value);
             throw new Error(`Cannot use ${className} "${stringifiedValue}" from another module or realm.
 
diff --git a/type/schema.js b/type/schema.js
index d340e68..faae3fa 100644
--- a/type/schema.js
+++ b/type/schema.js
@@ -216,9 +216,9 @@ class GraphQLSchema {
         );
 
       if (this._typeMap[typeName] !== undefined) {
-        throw new Error(
-          `Schema must contain uniquely named types but contains multiple types named "${typeName}".`,
-        );
+        // throw new Error(
+        //   `Schema must contain uniquely named types but contains multiple types named "${typeName}".`,
+        // );
       }
 
       this._typeMap[typeName] = namedType;
